src/main/java/org/interledger/codecs/psk/PskMessageBinaryCodec.java

```diff
@@ -0,0 +1,267 @@
+package org.interledger.codecs.psk;
+
+import org.interledger.Condition;
+import org.interledger.codecs.Codec;
+import org.interledger.codecs.CodecContext;
+import org.interledger.codecs.CodecException;
+import org.interledger.codecs.PskMessageCodec;
+import org.interledger.psk.PskEncryptionType;
+import org.interledger.psk.PskMessage;
+import org.interledger.psk.PskMessage.Header;
+
+import java.io.BufferedWriter;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+/**
+ * An implementation of {@link Codec} that reads and writes instances of {@link Condition}.
+ */
+public class PskMessageBinaryCodec implements PskMessageCodec {
+
+  private static final char CR = '\r';
+  private static final char LF = '\n';
+
+  /**
+   * Reads data of the input stream until a line feed character ('\n') is found or the input
+   * stream is exhausted. Returns the data interpreted as a *UTF-8* string. Note that lines ending
+   * with '\n' or '\r\n' are treated the same, the trailing '\r' will be removed if present.
+   *
+   * @return A UTF-8 encoded string read of the input stream.
+   */
+  private static String readLine(InputStream in) throws IOException {
+
+    boolean readCarriageReturn = false;
+
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream(256)) {
+      int byteValue;
+      while ((byteValue = in.read()) != -1) {
+
+        if (byteValue == CR) {
+          // Found CR, don't write it unless the next char is NOT a LF
+          readCarriageReturn = true;
+          continue;
+        }
+
+        if (byteValue == LF) {
+          break;
+        } else {
+          if (readCarriageReturn) {
+            // Not end of the line but we read a CR previously so write it before the current byte
+            bos.write(CR);
+            readCarriageReturn = false;
+          }
+        }
+        bos.write(byteValue);
```

Comment:
its equivalent because of the continues / breaks in the other if statements, but couldnt we neaten this up to

if (bytevalue == CR) {... continue;}
if (bytevalue == LF) {break;}
if (readCarriageReturn {...} else {bos.write(byteValue);}

we remove one level of nesting, and its clear that either we read a CR and write it back, or we write whatever byte we read.